crates/mcp-server/src/errors.rs

```diff
@@ -0,0 +1,71 @@
+use thiserror::Error;
+
+pub type BoxError = Box<dyn std::error::Error + Sync + Send>;
+
+#[derive(Error, Debug)]
+pub enum TransportError {
+    #[error("IO error: {0}")]
+    Io(#[from] std::io::Error),
+    
+    #[error("JSON serialization error: {0}")]
+    Json(#[from] serde_json::Error),
+    
+    #[error("Invalid UTF-8 sequence: {0}")]
+    Utf8(#[from] std::string::FromUtf8Error),
+    
+    #[error("Protocol error: {0}")]
+    Protocol(String),
+
+    #[error("Invalid message format: {0}")]
+    InvalidMessage(String),
+}
+
+#[derive(Error, Debug)]
+pub enum ServerError {
+    #[error("Transport error: {0}")]
+    Transport(#[from] TransportError),
+    
+    #[error("Service error: {0}")]
+    Service(String),
+    
+    #[error("Internal error: {0}")]
+    Internal(String),
+
+    #[error("Request timed out")]
+    Timeout(#[from] tower::timeout::error::Elapsed),
+}
+
+#[derive(Error, Debug)]
+pub enum RouterError {
+    #[error("Method not found: {0}")]
+    MethodNotFound(String),
+    
+    #[error("Invalid parameters: {0}")]
+    InvalidParams(String),
+    
+    #[error("Internal error: {0}")]
+    Internal(String),
+}
```

Comment:
What do you think of expanding the error typing from string? 

```
#[derive(Error, Debug)]
pub enum RouterError {
    // Instead of just a String, we keep the original error
    MethodNotFound {
        method: String,
        available_methods: Vec<String>
    },
    
    // We can include exactly what parameter was wrong
    InvalidParams {
        param_name: String,
        expected_type: String,
        received_value: String
    },
    
    // We can chain the original error that caused the problem
    Internal {
        message: String,
        #[source]
        cause: Box<dyn std::error::Error + Send + Sync>
    }
}
```

Currently this would only show the error message

Something like the above would show specific error messages, suggest fixes, log detailed error info, handle different errors differently which I think makes for a more pleasant debugging experience